python/GangaDirac/Lib/Backends/DiracBase.py

```diff
@@ -483,9 +520,30 @@ def debug(self):
             logger.error(result.get('Message', ''))
 
     @staticmethod
-    def _getStateTime(job, status):
+    def _bulk_updateStateTime( jobStateDict ):
+        """ This performs the same as the _getStateTime method but loops over a list of job ids within the DIRAC namespace (much faster)
```

Comment:
Yes, we probably need independent calls for now in a few other places. I'll review this once the PR is finished, debugging a few things atm